server/lib/models/model.go

```diff
@@ -61,27 +56,42 @@ func Init(db_addr, certs, host_addr string) error {
 	InitTopic()
 	InitReaction()
 	InitPost()
+	log.Printf("initialize db")
 	//create table according to model definition. TODO: how to do migration with roach?
     if err := dbm.CreateTablesIfNotExists(); err != nil {
     	log.Printf("CreateTablesIfNotExists: %v", err)
     	return err
     }
-    //initialize node object
-    dbm.node, _, err = NewNode(&dbm, host_addr)
-    if err != nil {
-    	log.Printf("NewNode: %v", err)
+    log.Printf("create databases")
+    if _, err := dbm.Exec("set database = channer;"); err != nil {
+    	log.Printf("exec set database: %v", err)
+    	return err;
+    }
+    if err := dbm.CreateIndex(); err != nil {
+    	if strings.Contains(err.Error(), "duplicate index name") {
+    		log.Printf("index already created: ignore error %v", err)
+    	} else {
+	    	log.Printf("CreateIndex: %v", err)
+    		return err    	
+    	}
+    }
+    log.Printf("create indexes")
+    //import data
+    if err := Import(data_path); err != nil {
+    	log.Printf("Import data: %v", err)    	
     	return err
     }
     log.Printf("model initialized")
+	utils.DumpMemUsage()
     return nil
 }
 
 func DBM() *Database {
 	return &dbm
 }
 
-func create_table(tmpl interface {}, name string, pkey_column string) *gorp.TableMap {
-	return dbm.AddTableWithNameAndSchema(tmpl, dbm.Name, name).SetKeys(false, pkey_column)
+func create_table(tmpl interface {}, name string, pkey_column ...string) *gorp.TableMap {
```

Comment:
__[gometalinter]__
* [golint] don't use underscores in Go names; func create_table should be createTable
* [golint] don't use underscores in Go names; func parameter pkey_column should be pkeyColumn